### DISCUSSION TITLE
Possibility of improving/trimming language list

### DISCUSSION THREAD

**Discussion by sircharlo** [sircharlo/meeting-media-manager#665] [Ideas] [answered]
Hey there @mtdvlpr :)

I found this interesting API today.

```
https://www.jw.org/en/library/magazines/json/filters/IssueYearViewsFilter/?contentLanguageFilter=en&pubFilter=w
https://www.jw.org/en/library/jw-meeting-workbook/json/filters/MWBViewsFilter/?contentLanguageFilter=en&pubFilter=mwb
```

Do you think this might be useful to perhaps add indicators to the language list if the w or mwb is available in that language? Or to disable fetching those pubs if unavailable?

Perhaps it's not worth it, but I thought it was an idea to put out there.

**Comment by mtdvlpr** [comment]
Nice one! I think it's worth it since I've seen several issues on Sentry regarding the mwb/w pubs not being available for their language. I'm very curious how you managed to find all those endpoints.

**Reply by sircharlo** [reply]
Yeah, those error are what led me to look for a potential solution. Digging around websites is fun :)

**Reply by sircharlo** [reply]
Basically, the usage would be: 

- Check for `mwb` availability: `https://www.jw.org/en/library/jw-meeting-workbook/json/filters/IssueYearViewsFilter/?contentLanguageFilter=[langSymbol]`
- Check for `w` availability: `https://www.jw.org/en/library/magazines/json/filters/MagazineViewsFilter/?contentLanguageFilter=[langSymbol]`

Compare results for `en` vs `ga`.

I'm thinking, once `langs` are loaded, each lang could be checked for w and mwb availability if not previously checked, and the result saved in langs.json. To prevent a ton of calls with every app launch. That way, every 3 months (or every cache clean), the app would know which languages it should not try to fetch w and mwb for.

**Reply by mtdvlpr** [reply]
Agreed. Altough you probably also don't want to check every language in a single go. Even when doing it simultaneously. So how do you propose to tackle that? Randomly checking x amount of langs each time the method is called?

**Reply by sircharlo** [reply]
True. Maybe check only when a language is selected by the user as the congregation's language?
Then, we could have a badge show up for the mw and we meetings depending on the results of the two calls.
Just thinking out loud here.

**Comment by mtdvlpr** [comment]
I found the issue for Irish. Turns out the watchtower is available, but under another pubSymbol. wqu instead of w

**Reply by sircharlo** [reply]
😮 well that's news for me

**Reply by mtdvlpr** [reply]
But it also seems that the querys need to be different for Irish...cause I'm getting an error for the query to fetch the document id

**Reply by mtdvlpr** [reply]
Turns out the JWPUB file for Irish doesn't have any multiMedia, so fixing it means M3 says weekend was successfully fetched and then not having any media for Saturday, so I'm gonna leave it as it is.

**Reply by sircharlo** [reply]
Did you look at the PDF file?

https://www.jw.org/ga/Leabharlann/iris%C3%AD/wqu202212/

`wqu` = Watchtower questions, from what I gather. In any case, not much we can do with that 🙂

**Reply by mtdvlpr** [reply]
Ah, okay. That explains it.

**Comment by mtdvlpr** [comment, marked as answer]
I made a PR for this :)
